docker/lib/dependabot/docker/tag.rb

```diff
@@ -5,7 +5,7 @@
 module Dependabot
   module Docker
     class Tag
-      BRANCH_WITH_BUILD = /(?:-[a-z]+)+-[0-9]+/
+      BRANCH_WITH_BUILD = /(?:(?:-[a-z]+)+-[0-9]+)+/
```

Comment:
Yeah... 😆.

What you point out seems like a bug in rubular (it claims to use Ruby 2.5.9, so pretty old). Scary because I use it quite a lot!

With irb on 3.1.4:

```
$ irb
irb(main):001:0> /(?:-[a-z]+)+-[0-9]+/.match("22-ea-7-windowsservercore-1809")
=> #<MatchData "-ea-7">
irb(main):002:0> /(?:(?:-[a-z]+)+-[0-9]+)+/.match("22-ea-7-windowsservercore-1809")
=> #<MatchData "-ea-7-windowsservercore-1809">
```

Reply:
> What you point out seems like a bug in rubular (it claims to use Ruby 2.5.9, so pretty old). Scary because I use it quite a lot!

Ah that is scary! Thanks for pointing it out